AdSecGH/Parameters/AdSecSection.cs

```diff
@@ -131,135 +182,128 @@ internal void CreatePreview(
         }
       }
 
-      var offs = Vector3d.Zero;
+      return flat;
+    }
 
-      if (offset != null) {
-        offs = new Vector3d(0, offset.Y.As(DefaultUnits.LengthUnitGeometry),
-          offset.Z.As(DefaultUnits.LengthUnitGeometry));
-      }
+    internal void AssignPreviewData(IPoint subComponentOffset = null) {
+      CreatePreview(out var profileData, out var reinforcementData, out var subData, subComponentOffset);
+      _profile = profileData.Profile;
+      _profileEdge = profileData.ProfileEdge;
+      _profileVoidEdges = profileData.ProfileVoidEdges;
+      _profileColour = profileData.ProfileColour;
+      _rebars = reinforcementData.Rebars;
+      _rebarEdges = reinforcementData.RebarEdges;
+      _linkEdges = reinforcementData.LinkEdges;
+      _rebarColours = reinforcementData.RebarColours;
+      _subProfiles = subData.SubProfiles;
+      _subEdges = subData.SubEdges;
+      _subVoidEdges = subData.SubVoidEdges;
+      _subColours = subData.SubColours;
```

Comment:
what I am thinking is do we even needs those if you created special structures? Shall we create fields for your `ProfilePreviewData`, `ReinforcementPreviewData` etc.. and ditch those?